core/modules/date/date.field.inc

```diff
@@ -557,3 +677,69 @@ function date_field_settings_form($field, $instance, $has_data) {
   module_load_include('inc', 'date', 'date.admin');
   return _date_field_settings_form($field, $instance, $has_data);
 }
+
+/**
+ * Custom callback to provide the widget settings form.
+ *
+ * @param array $instance
+ *   Field instance definition.
+ *
+ * @return array
+ *   Widget settings form structure.
+ */
+function _date_html5_field_widget_settings_form(array $instance) {
+  $settings = $instance['widget']['settings'];
+  $form = array();
+
+  $form['limits'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Date input limits'),
+    '#description' => t('Applies to start and end if enabled and is relative to the current date.'),
+  );
+  $form['limits']['mode'] = array(
+    '#type' => 'radios',
+    '#title' => t('Limits'),
+    '#options' => array(
+      'none' => t('No limit'),
+      'byyear' => t('By year'),
+    ),
+    '#default_value' => $settings['limits']['mode'],
+  );
+  $form['limits']['years_back'] = array(
+    '#type' => 'number',
+    '#title' => t('Starting year'),
+    '#min' => -9999,
+    '#max' => 0,
+    '#default_value' => $settings['limits']['years_back'],
+    '#description' => t('Empty value for no limit.'),
+    '#field_suffix' => t('from current year'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="instance[widget][settings][limits][mode]"]' => array('value' => 'byyear'),
+      ),
+    ),
+  );
+  $form['limits']['years_forward'] = array(
+    '#type' => 'number',
+    '#title' => t('Ending year'),
+    '#min' => 0,
+    '#max' => 9999,
+    '#default_value' => $settings['limits']['years_forward'],
+    '#description' => t('Empty value for no limit.'),
+    '#field_suffix' => t('from current year'),
+    '#states' => array(
+      'visible' => array(
+        ':input[name="instance[widget][settings][limits][mode]"]' => array('value' => 'byyear'),
+      ),
+    ),
+  );
+
+  $form['no_fieldset'] = array(
+    '#type' => 'checkbox',
+    '#title'  => t('Exclude wrapping fieldset'),
+    '#default_value' => !empty($settings['no_fieldset']),
+    '#description' => t('If using simple inputs such as a single textfield or the date popup, omitting the fieldset may provide simpler styling.'),
```

Comment:
```suggestion
    '#description' => t('If using simple inputs such as a single textfield or the date popup, omitting the fieldset may provide simpler styling.'),
    '#fieldset' => 'date_format',
```

This groups it into one of the two groupings for Date module.

![2022-08-28 12 15 06 bd1 lndo site f38a88fa0d21](https://user-images.githubusercontent.com/1071483/187084311-262a28e3-47ad-4913-9602-12beebaa3a5f.png)

Reply:
Hm... honestly, I don't think that the input limit (there are more options to come in a follow up) and the *display* (input(s) wrapped in fieldset or not) do belong to the same group.

Reply:
Maybe I'm biased, but this new settings form seems good to me:

![new-widget-settings](https://user-images.githubusercontent.com/842066/187146045-70505e4e-8f65-4fde-b65d-8426f77aad84.png)

Reply:
@indigoxela - I agree this looks good